src/Demo.mad

```diff
@@ -0,0 +1,78 @@
+import {} from 'String'
+import {} from 'Number'
+import IO from 'IO'
+
+import {getLocation} from './Browser'
+import {instigate} from './Main'
+
+hex :: String -> String
+hex = (r) => (
+  'hsla(' ++ r ++ ', 100%, 50%, 1)'
+)
+random :: Number -> Number
+random = (x) => #- Math.random() * x -#
+
+randomHex :: Number -> String
+randomHex = (_) => pipe(
+  random,
+  show,
+  hex
+)(360)
+
+
+setBackground = (color, x) => #-{
+  document.body.style.background = color
+  return x
+}-#
```

Comment:
It's always better to have type annotations for fenced stuff. I think we should emit compiler warnings in a few cases such as the body of a function or an assignment is fenced without type annotation:

```madlib
x = #- ... -#
f = (x) => #- ... -#
```